app/src/components/core/metaschema/metaSchema.ts

```diff
@@ -16,102 +16,146 @@ module app.core.metaschema {
         static fromJSON(json:any):Metaschema {
             var definitions:Definition[] = [];
 
-            // array contains all names of the elements in the root scope of the json, so they can be referenced by '#'.
-            var rootElements:string[] = [];
+            // array contains the names of all definitions
+            var definitionsNames:string[] = [];
 
             if (json.hasOwnProperty('definitions')) {
-                _.forOwn(json['definitions'], (value:{}, name:string) => {
-                    // we are only looking at object definitions (e.g. 'label' gets sorted out)
+                var definitionsRefs = json['anyOf'];
+                for (var i = 0; i < definitionsRefs.length; i++) {
+                    var name = definitionsRefs[i]['$ref'].substring(("#/definitions/").length);
```

Comment:
Provide a helper method that drops the `#/definitions` prefix, since this logic is used multiple times (for instance in the `cleanupDataschema` method)